pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/BooleanOrAggregationFunction.java

```diff
@@ -19,18 +19,21 @@
 
 package org.apache.pinot.core.query.aggregation.function;
 
+import com.google.common.base.Preconditions;
+import java.util.List;
 import org.apache.pinot.common.request.context.ExpressionContext;
 import org.apache.pinot.segment.spi.AggregationFunctionType;
 
 
 public class BooleanOrAggregationFunction extends BaseBooleanAggregationFunction {
 
-  public BooleanOrAggregationFunction(ExpressionContext expression) {
-    this(expression, false);
+  public BooleanOrAggregationFunction(List<ExpressionContext> arguments, boolean nullHandlingEnabled) {
+    super(verifyArguments(arguments), nullHandlingEnabled, BooleanMerge.OR);
   }
 
-  protected BooleanOrAggregationFunction(ExpressionContext expression, boolean nullHandlingEnabled) {
-    super(expression, nullHandlingEnabled, BooleanMerge.OR);
+  private static ExpressionContext verifyArguments(List<ExpressionContext> arguments) {
+    Preconditions.checkArgument(arguments.size() == 1, "BOOL_OR expects 1 argument, got: %s", arguments.size());
+    return arguments.get(0);
```

Comment:
(nit) this piece of code can potentially be abstracted out where each concrete function just passes it's name for the error message and the number of arguments it expects. 

May be put in `BaseSingleInputAggregationFunction` ?